examples/LearnHamiltonian.jl

```diff
@@ -0,0 +1,214 @@
+println("Optim training history",p_Optim)
+
+
+################################
+function loss3(p)
+      _HK = [0    0    p[2];
+             0    p[3] p[4];
+             p[2] p[4] p[5]]
+      _HS = [0  p[1]  0;
+            -p[1]  0  0;
+             0  0  0]
+      _L1 = [0.0 p[6] 0.0;
+            0.0 0.0  p[7];
+            0.0 0.0  0.0]
+      _L2 = [0.0 0.0 0.0;
+            0.0 p[8] 0.0;
+            0.0 0.0 2.0*p[8]]
+      _L_list = (_L1,_L2)
+      _LK,_LS,_LD = GLOQ.make_lindblad_operator(_HK,_HS,_L_list)
+
+      _problem = GLOQ.LindbladODEProblem(rho1_initial_real,rho1_initial_adj_imag,
+                                              _LK,_LS,_LD,
+                                               t_span[end];initial_type="density")
+      rho1_data = solve(_problem,saveat=t_span)
+      rho2_data = solve(remake(_problem, u0=rho2_initial), saveat=t_span)
+      rho3_data = solve(remake(_problem, u0=rho3_initial), saveat=t_span)
+      rho4_data = solve(remake(_problem, u0=rho4_initial), saveat=t_span)
+      rho5_data = solve(remake(_problem, u0=rho5_initial), saveat=t_span)
+
+      pop_data1 = GLOQ.get_population(Array(rho1_data[1:Int64(end/2),:]))
+      pop_data2 = GLOQ.get_population(Array(rho2_data[1:Int64(end/2),:]))
+      pop_data3 = GLOQ.get_population(Array(rho3_data[1:Int64(end/2),:]))
+      pop_data4 = GLOQ.get_population(Array(rho4_data[1:Int64(end/2),:]))
+      pop_data5 = GLOQ.get_population(Array(rho5_data[1:Int64(end/2),:]))
+
+      _loss = sum(abs2, pop_data1-noisy_data1)
+      _loss += sum(abs2, pop_data2-noisy_data2)
+      _loss += sum(abs2, pop_data3-noisy_data3)
+      _loss += sum(abs2, pop_data4-noisy_data4)
+      _loss += sum(abs2, pop_data5-noisy_data5)
+      return _loss
+end
+
+p3_initial = [1.0;1.0;0.25;0.1;0.1;0.5;0.5;0.5]
+@time p3_DEFlux = DiffEqFlux.sciml_train(loss3, p3_initial)
+@time p3_Optim = Optim.optimize(loss3, p3_initial)
+println("True p: ",p_true)
+println("DEFlux p: ",p3_DEFlux.u)
+println("Optim p: ",p3_Optim.minimizer)
+println(p3_DEFlux)
+println(p3_Optim)
+
+##########################################
+function test_auto_diff(p)
+      _HK = [0  0  p[2];
+             0  0  0;
+             p[2] 0  0]
+      _HS = [0  p[1]  0;
+            -p[1]  0  0;
+             0  0  0]
+      _L1 = [0.0 p[3] 0.0;
+            0.0 0.0  p[4];
+            0.0 0.0  0.0]
+      _L2 = [0.0 0.0 0.0;
+            0.0 p[5] 0.0;
+            0.0 0.0 2.0*p[5]]
+      _L_list = (_L1,_L2)
+      _LK,_LS,_LD = GLOQ.make_lindblad_operator(_HK,_HS,_L_list)
+      _problem = GLOQ.LindbladODEProblem(rho1_initial_real,rho1_initial_adj_imag,
+                                        _LK,_LS,_LD,
+                                         t_span[end];initial_type="density")
+      rho1_data = solve(_problem)
+#      return norm(Array(rho1_data[:,end]),2)^2 # AD works
+      pop_data1 = transpose(rho1_data[1:4:end,:])
+      return norm(pop_data1[:,end],2)
+end
+@time Zygote.gradient(test_auto_diff, p_true)
```

Comment:
I get: ERROR: 

LoadError: UndefVarError: norm not defined
Stacktrace:
 [1] _pullback
   @ ~/.julia/dev/GLOQ/examples/LearnHamiltonian.jl:211 [inlined]
 [2] _pullback(ctx::Zygote.Context, f::typeof(test_auto_diff), args::Matrix{Float64})
   @ Zygote ~/.julia/packages/Zygote/TaBlo/src/compiler/interface2.jl:0
 [3] _pullback(f::Function, args::Matrix{Float64})
   @ Zygote ~/.julia/packages/Zygote/TaBlo/src/compiler/interface.jl:34
 [4] pullback(f::Function, args::Matrix{Float64})
   @ Zygote ~/.julia/packages/Zygote/TaBlo/src/compiler/interface.jl:40
 [5] gradient(f::Function, args::Matrix{Float64})
   @ Zygote ~/.julia/packages/Zygote/TaBlo/src/compiler/interface.jl:75
 [6] top-level scope
   @ ./timing.jl:210
 [7] include(fname::String)
   @ Base.MainInclude ./client.jl:444
 [8] top-level scope
   @ REPL[16]:1
in expression starting at /Users/appelo/.julia/dev/GLOQ/examples/LearnHamiltonian.jl:213

Reply:
Just fix it.